tools/generator/src/DTO/BazelLabel.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+
+struct BazelLabel: Equatable, Hashable {
+    let repository: String
+    let package: String
+    let name: String
+}
+
+extension BazelLabel {
+    static let rootSeparator = "//"
+    static let nameSeparator = ":"
+    static let packageSeparator = "/"
+
+    init?(_ value: String) {
+        let rootParts = value.components(separatedBy: Self.rootSeparator)
+        guard rootParts.count == 2 else {
+            return nil
+        }
+
+        let repository = rootParts[0]
+        let packageAndNameParts = rootParts[1].components(separatedBy: Self.nameSeparator)
+
+        let package: String
+        let name: String
+        if packageAndNameParts.count == 2 {
+            package = packageAndNameParts[0]
+            name = packageAndNameParts[1]
+        } else if packageAndNameParts.count == 1 {
+            package = packageAndNameParts[0]
+            guard package != "" else {
+              return nil
+            }
+            let packageParts = package.components(separatedBy: Self.packageSeparator)
+            guard let lastPart = packageParts.last else {
+              return nil
+            }
+            name = lastPart
+        } else {
+            return nil
+        }
+
+        self.init(
+            repository: repository,
+            package: package,
+            name: name
+        )
+    }
+}
+
+extension BazelLabel: CustomStringConvertible {
+    var description: String {
+        return "\(repository)//\(package):\(name)"
+    }
+}
+
+extension BazelLabel: RawRepresentable {
+    init?(rawValue: String) {
+        self.init(rawValue)
+    }
+
+    var rawValue: String { "\(self)" }
+}
```

Comment:
Since we have a custom decoder (which I'm not sure we need if we have this), I don't think we need to/should conform to `RawRepresentable`.